File: backend/controllers/addToWishlist.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

// Construct an absolute path using path.join()
const parent=path.resolve(__dirname, '../../')
const absoluteFilePath = path.join(parent, 'product_sample.db');
// Create a database connection
<<<<<<< HEAD
<<<<<<< HEAD
const db = new sqlite3.Database('C:\Users\Rinu Joseph\Price-comparison-website\product_sample.db');
=======
const db = new sqlite3.Database(absoluteFilePath);
>>>>>>> 8b132d3833515654f3add6f915851c4a778d5480
=======
const db = new sqlite3.Database(absoluteFilePath);
>>>>>>> f91a40051ec59b2a316e30bb5cc360219aa72ad6

// Create WishlistProduct table
db.run(`CREATE TABLE IF NOT EXISTS WishlistProduct (
    productId INTEGER PRIMARY KEY,
    amazonLink TEXT,
    flipkartLink TEXT,
    cromaLink TEXT,
    amazonPrice REAL,
    flipkartPrice REAL,
    cromaPrice REAL,
    title TEXT,
    image TEXT
)`);

// Create WishlistEntry table
db.run(`CREATE TABLE  IF NOT EXISTS WishlistEntry (
    email TEXT,
    productId INTEGER,
    targetPrice REAL,
    FOREIGN KEY (productId) REFERENCES WishlistProduct(productId)
)`);
 function callback(row,title,image,email,targetPrice,res){
    
    db.run(`insert into WishlistProduct(amazonLink,flipkartLink,cromaLink,amazonPrice,flipkartPrice,cromaPrice,title,image) values(?,?,?,?,?,?,?,?);`,[row.a_url||null,row.f_url,row.c_url,row.a_price,row.f_price,row.c_price,title,image],function(err) {
        if (err) {
            console.log(err);
            return;
        }
        console.log(this.lastID);
        addToUser(this.lastID,email,targetPrice,res);
    
});

}
// Function to add a product to the wishlist
function addToWishlist(req,res) {

            a_link=req.body.a_link;
            f_link=req.body.f_link;
            c_link=req.body.c_link;
            email=req.body.email;
            productId=req.body.productId;
            targetPrice=req.body.targetPrice;

            if(f_link!=null){
                const sql = 'SELECT  *  FROM grouping1 WHERE f_url = ?';
                const params = [f_link];
                 db.get(sql, params, (err, row) => {
                        if (err) {
                            throw err;
                        }
                        title=row.f_title;
                        image=row.f_image;
                        callback(row,title,image,email,targetPrice,res);
                        });
                        
            }else if(a_link!=null){
                const sql = 'SELECT * FROM grouping1 WHERE a_url = ?';
                const params = [a_link];
                 db.get(sql, params, (err, row) => {
                        if (err) {
                            throw err;
                        }
                        title=row.a_title;
                        image=row.a_image;
                        callback(row,title,image,email,targetPrice,res);
                        });
            }else{
                
                const sql = 'SELECT * FROM grouping1 WHERE c_url = ?';
                const params = [c_link];
                 db.get(sql, params, (err, row) => {
                        if (err) {
                            throw err;
                        }
                        title=row.c_title;
                        image=row.c_image;
                        
                        callback(row,title,image,email,targetPrice,res);
                       
                        
                        });}
                        

            

            }

function addToUser(productId,email,targetPrice,res){
    db.run(`INSERT INTO WishlistEntry (email, productId, targetPrice) VALUES (?, ?, ?)`, 
                [email, productId, targetPrice], 
                function(err) {
                    if (err) {
                        console.error('Error adding entry to WishlistEntry:', err);
                        res.status(500).json({ error: 'cannot add to wishlist' });
                        return;
                    }
                    console.log('Product added to WishlistEntry table for email:', email);
                    res.status(201).json({ message: 'Added to WishList' });
                });
}
process.on('exit', () => db.close()); // Ensure database connection is closed when the script exits
module.exports={ addToWishlist}

File: addToWishlist.py
import sqlite3
from fastapi import FastAPI, HTTPException
from pydantic import BaseModel
from typing import Optional
app = FastAPI()

DATABASE_PATH = "product_sample.db"
class Wishlist(BaseModel):
    productId:int
    targetPrice:float
# Create a database connection
def create_connection():
    conn = sqlite3.connect(DATABASE_PATH)
    return conn

# Function to create WishlistProduct table
def create_product_table(conn):
    cursor = conn.cursor()
    cursor.execute('''
        CREATE TABLE IF NOT EXISTS WishlistProduct (
            productId INTEGER PRIMARY KEY,
            title TEXT,
        image TEXT,
        a_url TEXT,
        a_price TEXT,
        f_url TEXT,
        f_price TEXT,
        c_url TEXT,
        c_price TEXT
        )
    ''')
    conn.commit()

# Function to create WishlistEntry table
def create_wishlist_table(conn):
    cursor = conn.cursor()
    cursor.execute('''
        CREATE TABLE IF NOT EXISTS WishlistEntry (
            email TEXT,
            productId INTEGER,
            targetPrice text,
            FOREIGN KEY (productId) REFERENCES WishlistProduct(productId)
        )
    ''')
    conn.commit()

# Function to add a product to the wishlist
# @app.post("/add_to_wishlist/")
# async def add_to_wishlist(wish:WishList):
#     conn = create_connection()
#     create_product_table(conn)
#     create_wishlist_table(conn)  
    
#     cursor = conn.cursor()
#     cursor.execute('SELECT * FROM grouping WHERE productId=?',[wish.productId])
#     row = cursor.fetchone()
#     if not row:
#         raise HTTPException(status_code=404, detail="Product not found")
#     print(row)
#     title = row[0]  # Assuming title is the second column in grouping1 table
#     image = row[1]  # Assuming image is the third column in grouping1 table
    
#     cursor.execute("""insert into WishlistProduct(title ,
#         image ,
#         a_url ,
#         a_price ,
#         f_url ,
#         f_price,
#         c_url ,
#         c_price ) values(?,?,?,?,?,?,?,?)""",(title,image,row[2],row[3],row[4],row[5],row[6],row[7]))
#     product_id = cursor.lastrowid
#     conn.commit()
#     cursor.execute('''
#         INSERT INTO WishlistEntry (email, productId, targetPrice)
#         VALUES (?, ?, ?)
#     ''', (wish.email, product_id, wish.targetPrice))
#     conn.commit()
#     conn.close()

#     return {"message": "Added to WishList"}


